### PR TITLE
M2 1083 data screen timeline chart

### PR DESCRIPTION
**Pull request by DimitrySakalouski**

### CODE REVIEW

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+type Props = {
+  options: Array<TimelineChartOption>;
+  data: Array<ChartItem>;
+};
+
+const TimelineChart: FC<Props> = ({ data, options }) => {
+  const generateXAxisDots = () =>
+    Array.from(Array(7).keys()).map(item => ({ dot: item, value: 0 }));
+
+  const mappedData = useMemo(() => {
```
Comment (anq83) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:35]: Please add more types into your overall code of this component.
For example not clear what is mappedData

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    const dateFormat = 'yyyy dd MM';
+    const currentWeekDates = getCurrentWeekDates(dateFormat) as Array<string>;
+
+    const formattedDataDate = () =>
```
Comment (anq83) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:39]: formattedDataDate doesnt looks like function name.

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      data.map(dataItem => ({
```
Comment (anq83) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:40]: Please avoid using spreads here

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import { ChartItem } from '../types';
+
+type TimelineChartOption = {
+  name: string;
+  value: number;
+};
+
+type ChartRowItem = {
+  optionName: string;
+  timelines: Array<{ date: string; value?: number | null }>;
```
Comment (anq83) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:23]: Do not use strings indead of dates. Do not rely on dto-s, this is model layer

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+    const formattedDataDate = () =>
+      data.map(dataItem => ({
+        ...dataItem,
+        date: format(new Date(dataItem.date), dateFormat),
+      }));
+
+    const formattedDates = formattedDataDate();
+
+    const getValue = (option: TimelineChartOption, currentWeekDate: string) => {
```
Comment (anq83) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:47]: What is result type of this function? the same - add more types

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+          newDateItem.value === option.value
+        );
+      })?.value;
+    };
+
+    const getTimelineItems = (option: TimelineChartOption) => {
+      return currentWeekDates.map(currentWeekDate => {
+        return {
+          date: currentWeekDate,
+          value: !isNaN(Number(getValue(option, currentWeekDate)))
```
Comment (anq83) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:60]: add isNaN check into getValue function

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+            ? getValue(option, currentWeekDate)
+            : null,
+        };
+      });
+    };
+
+    const res: Array<ChartRowItem> = options.map(option => {
```
Comment (anq83) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:67]: rename res -> result

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    const getValue = (option: TimelineChartOption, currentWeekDate: string) => {
+      return formattedDates.find(newDateItem => {
+        return (
+          newDateItem.date === currentWeekDate &&
+          newDateItem.value === option.value
+        );
+      })?.value;
+    };
+
+    const getTimelineItems = (option: TimelineChartOption) => {
```
Comment (anq83) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:56]: Please consider moving the functions in the hook outside of the hook, in order to make hook code logic more clear

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+            style={{ data: { fill: colors.lighterGrey } }}
+            data={generateXAxisDots()}
+            x="dot"
+            y="value"
+            domain={[0, 6]}
+            size={5}
+          />
+
+          <VictoryAxis
+            tickValues={DAYS_OF_WEEK_NUMBERS}
```
Comment (anq83) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:100]: I'd prefer constants name like: DaysOfWeekNumbers i.o DAYS_OF_WEEK_NUMBERS

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/shared/ui/charts/TimelineChart/TimelineChart.tsx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+      return {
+        optionName: option.name,
+        timelines: getTimelineItems(option),
+      };
+    });
+
+    return res;
+  }, [data, options]);
+
+  const getScatterDots = (option: ChartRowItem) => {
```
Comment (BamMironov) [inline src/shared/ui/charts/TimelineChart/TimelineChart.tsx:77]: Please move it outside of the component scope

**Review by BamMironov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**